### PR TITLE
Skip "find_split_transpose" if split has more than one outputs

### PR DESCRIPTION
**Pull request by umangyadav**
The current matcher is doing an assert on the number of outputs but instead, it should skip it if the number of outputs is greater than 1. 

This matcher is doing 
```
input --> slice --> transpose 
    | 
     ---->slice --> transpose 
    | 
    ---->slice --> transpose 
```
to 
```
input --> transpose --> slice
                | 
                 ----> slice 
                | 
                ----> slice 

```
This is only useful if slice in original input graph has one output node (transpose in this case). If it has multiple outputs then doing transpose before the slice would change the layout for the slice's outputs (other than transpose) in original graph. 

Partially solves #2116

### CODE REVIEW

File: src/simplify_algebra.cpp [hunk 1]
```diff
@@ -1446,10 +1446,13 @@ struct find_split_transpose
         {
             return;
         }
+        if(std::any_of(split_outputs.begin(), split_outputs.end(), [](auto i) {
+               return i->outputs().size() != 1;
+           }))
+            return;
```
Comment (pfultz2) [inline src/simplify_algebra.cpp:1452]: I think can add `match::used_once()` to the matcher instead.

File: src/simplify_algebra.cpp [hunk 2]
```diff
@@ -1446,10 +1446,13 @@ struct find_split_transpose
...
 
         std::vector<instruction_ref> vec_trans(split_outputs.size());
         std::transform(split_outputs.begin(), split_outputs.end(), vec_trans.begin(), [](auto i) {
-            assert(i->outputs().size() == 1);
```
Comment (pfultz2) [inline src/simplify_algebra.cpp:1452]: We can keep the assert to ensure the matcher is working correctly.

File: src/simplify_algebra.cpp (hunk 1 above)
Comment (umangyadav) [reply src/simplify_algebra.cpp:1452]: I'll have to do it for all the slices, I dont' think i can do it inside matcher.

**Comment by kahmed10** [comment]
I think we need to add a unit test. I've seen this happen in distilgpt2 as well.

File: src/simplify_algebra.cpp (hunk 1 above)
Comment (pfultz2) [reply src/simplify_algebra.cpp:1452]: I think you could use `match::any_of[match::outputs()]` but that might be more complicated.

**Comment by pfultz2** [comment]
> I think we need to add a unit test. I've seen this happen in distilgpt2 as well.

Yea a unit test would be good.

**Review by pfultz2 — APPROVED** [review]

**Review by kahmed10 — APPROVED** [review]

**Pull request merged**